spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -734,6 +734,23 @@ class CometSparkSessionExtensions
       } else {
         var newPlan = transform(plan)
 
+        // if the plan cannot be run fully natively then explain why (when appropriate
+        // config is enabled)
+        if (CometConf.COMET_EXPLAIN_FALLBACK_ENABLED.get()) {
+          new ExtendedExplainInfo().extensionInfo(newPlan) match {
```

Comment:
Unrelated, but I feel `ExtendedExplainInfo` can be `object`

Reply:
Yes, I had the same thought

Reply:
It is possible that it has to be a class due to the way this integrates into Spark, but I am not sure.

Reply:
Just seeing this. I don't think Spark prevents us from this being an object. I'll take care of this. https://github.com/apache/datafusion-comet/issues/452